src/cclib/parser/qchemparser.py

```diff
@@ -154,6 +154,140 @@ def after_parsing(self):
                     newbfname = '{}{}'.format(bfcounts[bfname], bfname)
                     self.aonames[bfindex] = '_'.join([atomname, newbfname])
 
+    def parse_charge_section(self, inputfile, chargetype):
+        """Parse the population analysis charge block."""
+        self.skip_line(inputfile, 'blank')
+        line = next(inputfile)
+        has_spins = False
+        if 'Spin' in line:
+            if not hasattr(self, 'atomspins'):
+                self.atomspins = dict()
+            has_spins = True
+            spins = []
+        self.skip_line(inputfile, 'dashes')
+        if not hasattr(self, 'atomcharges'):
+            self.atomcharges = dict()
+        charges = []
+        line = next(inputfile)
+
+        while list(set(line.strip())) != ['-']:
+            elements = line.split()
+            charge = self.float(elements[2])
+            charges.append(charge)
+            if has_spins:
+                spin = self.float(elements[3])
+                spins.append(spin)
+            line = next(inputfile)
+
+        self.atomcharges[chargetype] = numpy.array(charges)
+        if has_spins:
+            self.atomspins[chargetype] = numpy.array(spins)
+
+    @staticmethod
```

Comment:
Yes. But modifying a passed object is not very Pythonic. Is there any reason not to create it in the method and return it?